parser-typechecker/src/Unison/PrettyPrintEnv.hs

```diff
@@ -54,9 +56,14 @@ termName env r =
 typeName :: PrettyPrintEnv -> Reference -> HashQualified Name
 typeName env r =
   case types env r of
-    Nothing -> HQ.take todoHashLength (HQ.fromReference r) 
+    Nothing -> HQ.take todoHashLength (HQ.fromReference r)
     Just name -> HQ'.toHQ name
 
+-- | Find a name for this referent (or embedded reference) in either the types or terms.
+typeOrTermName :: PrettyPrintEnv -> Referent -> HashQualified Name
+typeOrTermName env r =
+  maybe (HQ.take todoHashLength $ HQ.fromReferent r) HQ'.toHQ ((Referent'.toTermReference r >>= types env) <|> terms env r)
```

Comment:
Ok, I see it is doing maybe what you intended, but it breaks the rule of Referent:
```
-- | Specifies a term.
--
-- Either a term 'Reference', a data constructor, or an effect constructor.
--
-- Slightly odd naming. This is the "referent of term name in the codebase",
-- rather than the target of a Reference.

-- | When @Ref'@ then @r@ represents a term.
--
-- When @Con'@ then @r@ is a type declaration.
data Referent' r = Ref' r | Con' r ConstructorId ConstructorType
  deriving (Show, Ord, Eq, Functor, Generic)
```

Reply:
This is an issue which I believe comes from our lack of ability to distinguish between the type of dependency in our dependents index.
When I ask for dependencies I just get a bunch of References without knowing which are types and which are terms, so I'm currently just combining them all together and seeing what sticks.

Ideally, I think the DB would have more information and would return something akin to a LabeledDependency.

Options for now might be to:
1. Add some sort of way to check what a Reference points to as a method on the Codebase
2. Have `getDependencies` join in the object-type in SQLite, then return a Labeled Dependency which is either a TermReference or TypeReference (but we still have to do extra work to get Constructors)
3. Keep what we have, even though it's less precise and blurs boundaries.

I think I'd lean towards 2.